### PR TITLE
add cmdlets Get/Start/convert-SynapseVulnerabilityAssessmentScan

### PR DESCRIPTION
**Pull request by zesluo**
## Description

https://github.com/Azure/azure-powershell/issues/13731

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by wyunchi-ms** [comment]
Hi @zesluo , another PR is merged first, could you please help to resolve the conflicts?

File: src/Synapse/Synapse/Commands/ManagementCommands/SqlPool/ConvertAzureSynapseVulnerabilityAssessmentScan.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+{
+    [Cmdlet("Convert", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + SynapseConstants.SynapsePrefix + SynapseConstants.SqlPool + SynapseConstants.VulnerabilityAssessmentScan,
+        SupportsShouldProcess = true)]
+    [OutputType(typeof(PSVulnerabilityAssessmentScanExportModel))]
+    public class ConvertAzureSynapseVulnerabilityAssessmentScan : SynapseManagementCmdletBase
+    {
+        private const string ConvertByVulnerabilityAssessmentScanRecordModelParameterSet = "ConvertByVulnerabilityAssessmentScanRecordModelParameterSet";
+        private const string ConvertByNameParameterSet = "ConvertByNameParameterSet";
+ 
+        [Parameter(Mandatory = false, HelpMessage = HelpMessages.ResourceGroupName)]
```
Comment (idear1203) [inline src/Synapse/Synapse/Commands/ManagementCommands/SqlPool/ConvertAzureSynapseVulnerabilityAssessmentScan.cs:19]: >[Parameter(Mandatory = false, HelpMessage = HelpMessages.ResourceGroupName)] [](start = 8, length = 76)

Do we need to add "ValueFromPipelineByPropertyName = true" for `ResourceGroupName`?

File: src/Synapse/Synapse/Models/SynapseAnalyticsManagementClient.cs [hunk 2]
```diff
@@ -290,12 +294,75 @@ internal SqlPool GetSqlPool(string resourceGroupName, string workspaceName, stri
...
+                  {
+                      Code = scanError.Code,
+                      Message = scanError.Message
+                  }).ToList(),
+                ScanResultsLocationPath = scanRecord.StorageContainerPath,
+                NumberOfFailedSecurityChecks = scanRecord.NumberOfFailedSecurityChecks
+            };
+        }
+
+
```
Comment (idear1203) [inline src/Synapse/Synapse/Models/SynapseAnalyticsManagementClient.cs:365]: Remove one empty line

File: src/Synapse/Synapse/Models/PSVulnerabilityAssessmentScanExportModel.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        /// Gets or sets the Workspace name
+        /// </summary>
+        public string WorkspaceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the SqlPool name
+        /// </summary>
+        public string SqlPoolName { get; set; }
+
+
```
Comment (idear1203) [inline src/Synapse/Synapse/Models/PSVulnerabilityAssessmentScanExportModel.cs:27]: remove one empty line

File: src/Synapse/Synapse/help/Az.Synapse.md [hunk 4]
```diff
@@ -77,6 +77,15 @@ This feature is in a limited preview, initially accessible only to certain subsc
...
 Gets a Synapse Analytics SQL pool.
 
+### [Get-AzSynapseSqlPoolVulnerabilityAssessmentScanRecord](Get-AzSynapseSqlPoolVulnerabilityAssessmentScanRecord.md)
+Gets a Synapse Analytics SQL pool vulnerability assessment scan record.
+
+### [Convert-AzSynapseVulnerabilityAssessmentScan](Convert-AzSynapseVulnerabilityAssessmentScan.md)
+Converts to a Synapse Analytics SQL pool vulnerability assessment scan record.
+
+### [Start-AzSynapseVulnerabilityAssessmentScanInitialziation](Start-AzSynapseVulnerabilityAssessmentScanInitialziation.md)
+Start a Synapse Analytics SQL pool vulnerability assessment scan record.
```
Comment (idear1203) [inline src/Synapse/Synapse/help/Az.Synapse.md:87]: Do you modify this file manually? Usually we use the `Update-MarkdownHelpModule` command

Comment (zesluo) [reply src/Synapse/Synapse/help/Az.Synapse.md:87]: Got it, thanks. Yeah, I modified it manually

**Comment by zesluo** [comment]
> Hi @zesluo , another PR is merged first, could you please help to resolve the conflicts?

sure, thanks

File: src/Synapse/Synapse/Commands/ManagementCommands/SqlPool/StartAzureSynapseVulnerabilityAssessmentScan.cs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Synapse
+{
+    [Cmdlet("Start", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + SynapseConstants.SynapsePrefix + SynapseConstants.SqlPool + SynapseConstants.VulnerabilityAssessmentScanInitialziation, DefaultParameterSetName = StartSqlPoolObjectParameterSet
+, SupportsShouldProcess = true),OutputType(typeof(PSVulnerabilityAssessmentScanRecordModel))]
+    [OutputType(typeof(PSVulnerabilityAssessmentScanRecordModel))]
+    public class StartAzureSynapseSqlPoolVulnerabilityAssessmentScanInitialziation : SynapseManagementCmdletBase
```
Comment (idear1203) [inline src/Synapse/Synapse/Commands/ManagementCommands/SqlPool/StartAzureSynapseVulnerabilityAssessmentScan.cs:16]: >StartAzureSynapseSqlPoolVulnerabilityAssessmentScanInitialziation [](start = 17, length = 65)

Start-AzSynapseSqlPoolVulnerabilityAssessmentScan? Why do you add an Initialziation suffix?

File: src/Synapse/Synapse/Commands/ManagementCommands/SqlPool/GetAzureSynapseSqlPoolVulnerabilityAssessmentScanRecord.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+            ICollection<PSVulnerabilityAssessmentScanRecordModel> results;
+
+            bool getScanRecordBySpecificScanId = MyInvocation.BoundParameters.ContainsKey(nameof(ScanId));
+            string scanId = ScanId;
+
+            if (getScanRecordBySpecificScanId)
+            {
+                results = new List<PSVulnerabilityAssessmentScanRecordModel>
+                {
```
Comment (idear1203) [inline src/Synapse/Synapse/Commands/ManagementCommands/SqlPool/GetAzureSynapseSqlPoolVulnerabilityAssessmentScanRecord.cs:71]: Why put the results in a collection?

File: src/Synapse/Synapse/Common/ResourceTypes.cs [hunk 7]
```diff
@@ -5,6 +5,7 @@ public static class ResourceTypes
         public const string Workspace = "Microsoft.Synapse/workspaces";
         public const string SparkPool = "Microsoft.Synapse/workspaces/bigDataPools";
         public const string SqlPool = "Microsoft.Synapse/workspaces/sqlPools";
+        public const string ScanId = "Microsoft.Synapse/workspaces/sqlPools/scanId";
```
Comment (idear1203) [inline src/Synapse/Synapse/Common/ResourceTypes.cs:8]: >ScanId [](start = 28, length = 6)

Where is it used?

File: src/Synapse/Synapse/Az.Synapse.psd1 [hunk 8]
```diff
@@ -121,7 +121,9 @@ CmdletsToExport = 'Get-AzSynapseSparkJob', 'Stop-AzSynapseSparkJob',
                'Stop-AzSynapseTrigger', 'Get-AzSynapseTriggerRun', 
                'Set-AzSynapseDataset', 'Get-AzSynapseDataset',
                'Remove-AzSynapseDataset', 'Set-AzSynapseDataFlow',
-               'Get-AzSynapseDataFlow', 'Remove-AzSynapseDataFlow'
+               'Get-AzSynapseDataFlow', 'Remove-AzSynapseDataFlow',
+               'Get-AzSynapseSqlPoolVulnerabilityAssessmentScanRecord','Convert-AzSynapseSqlPoolVulnerabilityAssessmentScan',
+               'Start-AzSynapseSqlPoolVulnerabilityAssessmentScanInitialziation'
```
Comment (idear1203) [inline src/Synapse/Synapse/Az.Synapse.psd1:126]: >Initialziation [](start = 65, length = 14)

typo

File: src/Synapse/Synapse/help/Az.Synapse.md (hunk 4 above)
Comment (idear1203) [reply src/Synapse/Synapse/help/Az.Synapse.md:114]: Please use `Update-MarkdownHelpModule` to  update the entries again. These entries should be in alphabetical order.

---
In reply to: [542156034](https://github.com/Azure/azure-powershell/pull/13690#discussion_r542156034) [](ancestors = 542156034)

File: src/Synapse/Synapse/Models/PSVulnerabilityAssessmentScanExportModel.cs [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+using System.Text;
+using Microsoft.WindowsAzure.Commands.Common.Attributes;
+
+namespace Microsoft.Azure.Commands.Synapse.Models
+{
+
+    public class PSVulnerabilityAssessmentScanExportModel
+    {
+
+        /// <summary>
```
Comment (idear1203) [inline src/Synapse/Synapse/Models/PSVulnerabilityAssessmentScanExportModel.cs:13]: remove this empty line.

File: src/Synapse/Synapse/Commands/ManagementCommands/SqlPool/StartAzureSynapseVulnerabilityAssessmentScan.cs (hunk 5 above)
Comment (idear1203) [reply src/Synapse/Synapse/Commands/ManagementCommands/SqlPool/StartAzureSynapseVulnerabilityAssessmentScan.cs:16]: ping on this..

---
In reply to: [543111306](https://github.com/Azure/azure-powershell/pull/13690#discussion_r543111306) [](ancestors = 543111306)

**Comment by idear1203** [comment]
Forget to update Az.Synapse.psd1?

**Comment by msJinLei** [comment]
@zesluo 
Please fix the failed test cases. It seems that you need to re-record the case.

```
2021-02-01T13:05:25.1446488Z   [xUnit.net 00:00:13.29]     Microsoft.Azure.Commands.Synapse.Test.ScenarioTests.VulnerabilityAssessmentTests.TestVulnerabilityAssessmentScanRecordGetListTest [FAIL]
2021-02-01T13:05:25.3475282Z   [xUnit.net 00:00:13.43]     Microsoft.Azure.Commands.Synapse.Test.ScenarioTests.SparkJobTests.TestSynapseSparkJob [SKIP]
2021-02-01T13:05:25.7037659Z     X Microsoft.Azure.Commands.Synapse.Test.ScenarioTests.VulnerabilityAssessmentTests.TestVulnerabilityAssessmentScanRecordGetListTest [443ms]
2021-02-01T13:05:25.7168445Z EXEC : error Message:  [D:\a\1\s\build.proj]
2021-02-01T13:05:25.7170257Z      System.Management.Automation.ActionPreferenceStopException : The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Unable to find a matching HTTP request for URL 'POST /subscriptions/051ddeca-1ed6-4d8b-ba6f-1ff561e5f3b3/resourceGroups/sql-va-cmdlet-test-rgps5651/providers/Microsoft.Synapse/workspaces/sqlvawsps5651/sqlPools/sqlvapoolps5651/vulnerabilityAssessments/default/scans/20210201_130525/initiateScan?api-version=2019-06-01-preview'. Calling method Item().
2021-02-01T13:05:25.7171541Z     Stack Trace:
2021-02-01T13:05:25.7172075Z        at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
2021-02-01T13:05:25.7172722Z      at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
2021-02-01T13:05:25.7173652Z      at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
2021-02-01T13:05:25.7174453Z      at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-01T13:05:25.7176016Z      at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-01T13:05:25.7176805Z      at System.Management.Automation.PowerShell.CoreInvoke[TOutput](IEnumerable input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-01T13:05:25.7177498Z      at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
2021-02-01T13:05:25.7178087Z      at System.Management.Automation.PowerShell.Invoke()
2021-02-01T13:05:25.7178890Z      at Microsoft.WindowsAzure.Commands.ScenarioTest.EnvironmentSetupHelper.RunPowerShellTest(String[] scripts) in D:\a\1\s\tools\ScenarioTest.ResourceManager\EnvironmentSetupHelper.cs:line 573
2021-02-01T13:05:25.7180049Z      at Microsoft.Azure.Commands.Synapse.Test.ScenarioTests.SynapseTestBase.RunPsTestWorkflow(Func`1 scriptBuilder, Action cleanup, String callingClassType, String mockName) in D:\a\1\s\src\Synapse\Synapse.Test\ScenarioTests\SynapseTestBase.cs:line 121
2021-02-01T13:05:25.7181271Z      at Microsoft.Azure.Commands.Synapse.Test.ScenarioTests.SynapseTestBase.RunPsTest(XunitTracingInterceptor logger, String[] scripts) in D:\a\1\s\src\Synapse\Synapse.Test\ScenarioTests\SynapseTestBase.cs:line 74
2021-02-01T13:05:25.7182400Z      at Microsoft.Azure.Commands.Synapse.Test.ScenarioTests.VulnerabilityAssessmentTests.TestVulnerabilityAssessmentScanRecordGetListTest() in D:\a\1\s\src\Synapse\Synapse.Test\ScenarioTests\VulnerabilityAssessmentTests.cs:line 43
2021-02-01T13:05:25.7183230Z     Standard Output Messages:
2021-02-01T13:05:25.7359820Z    Test-VulnerabilityAssessmentScanRecordGetListTest
2021-02-01T13:05:25.7372766Z    Caught Exception: System.Management.Automation.ActionPreferenceStopException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Unable to find a matching HTTP request for URL 'POST /subscriptions/051ddeca-1ed6-4d8b-ba6f-1ff561e5f3b3/resourceGroups/sql-va-cmdlet-test-rgps5651/providers/Microsoft.Synapse/workspaces/sqlvawsps5651/sqlPools/sqlvapoolps5651/vulnerabilityAssessments/default/scans/20210201_130525/initiateScan?api-version=2019-06-01-preview'. Calling method Item().
2021-02-01T13:05:25.7375064Z       at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
2021-02-01T13:05:25.7376428Z       at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
2021-02-01T13:05:25.7434551Z       at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
2021-02-01T13:05:25.7435634Z       at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-01T13:05:25.7436567Z       at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-01T13:05:25.7437444Z       at System.Management.Automation.PowerShell.CoreInvoke[TOutput](IEnumerable input, PSDataCollection`1 output, PSInvocationSettings settings)
2021-02-01T13:05:25.7438234Z       at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
2021-02-01T13:05:25.7438877Z       at System.Management.Automation.PowerShell.Invoke()
2021-02-01T13:05:25.7439745Z       at Microsoft.WindowsAzure.Commands.ScenarioTest.EnvironmentSetupHelper.RunPowerShellTest(String[] scripts) in D:\a\1\s\tools\ScenarioTest.ResourceManager\EnvironmentSetupHelper.cs:line 573
2021-02-01T13:05:25.7441460Z    Message: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Unable to find a matching HTTP request for URL 'POST /subscriptions/051ddeca-1ed6-4d8b-ba6f-1ff561e5f3b3/resourceGroups/sql-va-cmdlet-test-rgps5651/providers/Microsoft.Synapse/workspaces/sqlvawsps5651/sqlPools/sqlvapoolps5651/vulnerabilityAssessments/default/scans/20210201_130525/initiateScan?api-version=2019-06-01-preview'. Calling method Item().
2021-02-01T13:05:25.7443296Z EXEC : PowerShell error Record: Unable to find a matching HTTP request for URL 'POST /subscriptions/051ddeca-1ed6-4d8b-ba6f-1ff561e5f3b3/resourceGroups/sql-va-cmdlet-test-rgps5651/providers/Microsoft.Synapse/workspaces/sqlvawsps5651/sqlPools/sqlvapoolps5651/vulnerabilityAssessments/default/scans/20210201_130525/initiateScan?api-version=2019-06-01-preview'. Calling method Item(). [D:\a\1\s\build.proj]
2021-02-01T13:05:25.7445066Z   Exception:System.Collections.Generic.KeyNotFoundException: Unable to find a matching HTTP request for URL 'POST /subscriptions/051ddeca-1ed6-4d8b-ba6f-1ff561e5f3b3/resourceGroups/sql-va-cmdlet-test-rgps5651/providers/Microsoft.Synapse/workspaces/sqlvawsps5651/sqlPools/sqlvapoolps5651/vulnerabilityAssessments/default/scans/20210201_130525/initiateScan?api-version=2019-06-01-preview'. Calling method Item().
2021-02-01T13:05:25.7446229Z       at Microsoft.Azure.Test.HttpRecorder.Records.get_Item(String key)
2021-02-01T13:05:25.7446935Z       at Microsoft.Azure.Test.HttpRecorder.HttpMockServer.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
2021-02-01T13:05:25.7447717Z       at System.Net.Http.HttpMessageInvoker.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
2021-02-01T13:05:25.7448643Z       at System.Net.Http.HttpClient.SendAsync(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationToken cancellationToken)
2021-02-01T13:05:25.7449442Z       at System.Net.Http.HttpClient.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
2021-02-01T13:05:25.7450452Z       at Microsoft.Azure.Management.Synapse.SqlPoolVulnerabilityAssessmentScansOperations.BeginInitiateScanWithHttpMessagesAsync(String resourceGroupName, String workspaceName, String sqlPoolName, String scanId, Dictionary`2 customHeaders, CancellationToken cancellationToken)
2021-02-01T13:05:25.7451695Z       at Microsoft.Azure.Management.Synapse.SqlPoolVulnerabilityAssessmentScansOperations.InitiateScanWithHttpMessagesAsync(String resourceGroupName, String workspaceName, String sqlPoolName, String scanId, Dictionary`2 customHeaders, CancellationToken cancellationToken)
2021-02-01T13:05:25.7452906Z       at Microsoft.Azure.Management.Synapse.SqlPoolVulnerabilityAssessmentScansOperationsExtensions.InitiateScanAsync(ISqlPoolVulnerabilityAssessmentScansOperations operations, String resourceGroupName, String workspaceName, String sqlPoolName, String scanId, CancellationToken cancellationToken)
2021-02-01T13:05:25.7454098Z       at Microsoft.Azure.Management.Synapse.SqlPoolVulnerabilityAssessmentScansOperationsExtensions.InitiateScan(ISqlPoolVulnerabilityAssessmentScansOperations operations, String resourceGroupName, String workspaceName, String sqlPoolName, String scanId)
2021-02-01T13:05:25.7455410Z       at Microsoft.Azure.Commands.Synapse.Models.SynapseAnalyticsManagementClient.StartVulnerabilityAssessmentScan(String resourceGroup, String workspaceName, String sqlPoolName, String scanId) in D:\a\1\s\src\Synapse\Synapse\Models\SynapseAnalyticsManagementClient.cs:line 1540
2021-02-01T13:05:25.7456751Z       at Microsoft.Azure.Commands.Synapse.StartAzureSynapseSqlPoolVulnerabilityAssessmentScan.ExecuteCmdlet() in D:\a\1\s\src\Synapse\Synapse\Commands\ManagementCommands\SqlPool\StartAzureSynapseSqlPoolVulnerabilityAssessmentScan.cs:line 78
2021-02-01T13:05:25.7457733Z       at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
2021-02-01T13:05:25.7458356Z   Details:
2021-02-01T13:05:25.7459155Z   Script Stack Trace: at Test-VulnerabilityAssessmentScanRecordGetListTest, D:\a\1\s\src\Synapse\Synapse.Test\bin\Debug\netcoreapp2.1\ScenarioTests\VulnerabilityAssessmentTests.ps1: line 186
2021-02-01T13:05:25.7460042Z    at <ScriptBlock>, <No file>: line 1
```

**Review by wyunchi-ms — APPROVED** [review]

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**